server/controllers/report.js

```diff
@@ -158,10 +166,74 @@ function deleteArchived(req, res, next) {
     .done();
 }
 
+// TODO(@jniles) - translate these emails into multiple languages
+const REPORT_EMAIL =
+`Hello!
+
+Please find the attached report "%filename%" produced by %user% on %date%.
+
+This email was requested by %requestor%.
+
+Thanks,
+bhi.ma
+`;
+
+// this is a really quick and lazy templating scheme
+const template = (str, values) => {
+  return Object.keys(values).reduce((formatted, key) =>
+    formatted.replace(`%${key}%`, values[key]), str);
+};
+
+
+/**
+ * @function emailArchived
+ *
+ * @description
+ * Emails an archived report to an email address provided in the "to" field.
+ */
+function emailArchived(req, res, next) {
+  const { uuid } = req.params;
+  const { address } = req.body;
+
+  debug(`#emailArchived(): Received email request for ${address}.`);
+
+  lookupArchivedReport(uuid)
+    .then(report => {
+      debug(`#emailArchived(): sending ${report.label} to ${address}.`);
```

Comment:
The re is a difference between `debug` and `winston`?

Reply:
Good question!  Winston uses "error levels" (winston.log(), winston.error(), winston.warn()), which debug simply namespaces logs by module.  It's like having `console.log()` statements that you can turn on and off.

This is advantageous because you can turn all logs off except the ones you care about, or you can filter logs based on what module it came from.  For example, here we log only email logs by using `DEBUG=mailer` and it will only print the `debug()` calls that are in the `mailer` module.  That way, if there are errors in the email script, we will be able to easily track them down in production by either filtering through the logs or turning off all logs except the email logs.

[This issue](https://github.com/IMA-WorldHealth/bhima-2.X/issues/1519) gives more of a detailed overview about why debug() works better for our needs than winston.

Reply:
Thank you for the explanation